cmd2/utils.py

```diff
@@ -55,3 +58,89 @@ def namedtuple_with_defaults(typename, field_names, default_values=()):
     T.__new__.__defaults__ = tuple(prototype)
     return T
 
+def namedtuple_with_two_defaults(typename, field_names, default_values=('', '')):
+    """Wrapper around namedtuple which lets you treat the last value as optional.
+
+    :param typename: str - type name for the Named tuple
+    :param field_names: List[str] or space-separated string of field names
+    :param default_values: (optional) 2-element tuple containing the default values for last 2 parameters in named tuple
+                            Defaults to an empty string for both of them
+    :return: namedtuple type
+    """
+    T = collections.namedtuple(typename, field_names)
+    # noinspection PyUnresolvedReferences
+    T.__new__.__defaults__ = default_values
+    return T
+
+def cast(current, new):
+    """Tries to force a new value into the same type as the current when trying to set the value for a parameter.
+
+    :param current: current value for the parameter, type varies
+    :param new: str - new value
+    :return: new value with same type as current, or the current value if there was an error casting
+    """
+    typ = type(current)
+    if typ == bool:
+        try:
+            return bool(int(new))
+        except (ValueError, TypeError):
+            pass
+        try:
+            new = new.lower()
+        except AttributeError:
+            pass
+        if (new == 'on') or (new[0] in ('y', 't')):
+            return True
+        if (new == 'off') or (new[0] in ('n', 'f')):
+            return False
+    else:
+        try:
+            return typ(new)
+        except (ValueError, TypeError):
+            pass
+    print("Problem setting parameter (now %s) to %s; incorrect type?" % (current, new))
+    return current
+
+def which(editor: str) -> Optional[str]:
+    import subprocess
```

Comment:
Maybe add a docstring comment?  I know the original probably didn't have one, but now that it is separated from its immediate use it might be helpful.